src/transformation/mesh_intersection/mesh_intersection.rs

```diff
@@ -243,7 +271,10 @@ pub fn intersect_meshes_with_tolerances(
     let vertices: Vec<_> = vertices.iter().map(|p| Point3::from(p.point)).collect();
 
     if !topology_indices.is_empty() {
-        Ok(Some(TriMesh::new(vertices, topology_indices)?))
+        Ok(Some(TriMesh::new(
+            vertices,
+            topology_indices.into_values().collect(),
```

Comment:
Very scared of the non-determinism of the hashmap.

Reply:
Cosnidering we even tolerated an n^2 algorithm elsewhere. I wold want to suggest switching to a BTreeMap, or at least benchmark to ensure that the performance gains justify a hashmap.